### PR TITLE
Refactor deployment modules

### PR DESCRIPTION
**Pull request by bilbof**
Changes:

- Uses tfvars: makes it easier to manage deployments to multiple environments.
- Removes duplicated configuration between the govuk-test and application deployment modules.
- Task definitions won't be created by the govuk module, so we'll need to run additional `terraform apply`s to bring up the apps.

Note: This Pull Request is most easily reviewed commit by commit.

New directory structure:

```
├── apps
│   ├── common.tf
│   ├── outputs.tf
│   ├── content-store
│   │   ├── common.tf -> ../common.tf
│   │   ├── main.tf
│   │   ├── outputs.tf -> ../outputs.tf
│   │   └── variables.tf -> ../variables.tf
│   └── variables.tf
├── govuk-test
│   ├── README.md
│   ├── main.tf
│   └── variables.tf
└── variables
    └── test
        ├── apps.tfvars
        ├── common.tfvars
        └── infrastructure.tfvars
```

This introduces a new directory: `terraform/deployments/variables`
which will contain the variable definitions (.tfvar) files that should
be called when applying terraform for an app task definition or for
the infrastructure.

The new workflows:

- Applying the infrastructure
  ```shell
  $ cd terraform/deployments/govuk-test
  $ terraform apply -var-file=../variables/test/common.tfvars \
     -var-file=../variables/test/infrastructure.tfvars
  ```
- Applying the application
  ```shell
  $ cd terraform/deployments/apps/frontend
  $ terraform apply -var-file=../../variables/test/common.tfvars \
     -var-file=../../variables/test/apps.tfvars \
     -var=image_tag=release_123
  ```

The variables/test/common.tfvars file contains variables common to the test
environment deployments.

Should we want to create a new environment we would need to duplicate these
tfvars files for another environment. When we add extra environments we may
find that we can introduce a cross-environment common.tfvars file.

Story: https://trello.com/c/2N5OLEHg/266-refactor-terraform-deployments

### CODE REVIEW

**Review by sengi — DISMISSED** [review]
I think this is a big win - gets rid of a whole layer of boilerplate modules and I'm hopeful it'll be Good Enough™️ for us not to need to open the whole putting-another-config-management-layer-on-top can of worms 🤞 (or splitting out task config management into a different system and figuring out an interface between Terraform and it)

I've really only had a quick glance through and not gone line by line or anything but LGTM in terms of the overall picture. (I don't want to discourage other reviewers from looking more closely or suggesting improvements though!)

**Review by fredericfran-gds — COMMENTED** [review]
In general, it looks good. Thank you for doing this. It seems saner now when adding an app though I have not done so yet. 

It would be useful if you could improve on the README.md on how to add a new app to the platform. I believe it is essential because we need to ensure it is a developer-friendly process and not confusing.

You can also add the deployment commands too even though we will be relying on Concourse to deploy the infrastructure.

**Comment by bilbof** [comment]
@fredericfran-gds thank you! I have added two documents

- Adding an app
- Applying Terraform

Please let me know what you think.

File: docs/adding-an-app.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+   ```sh
+   $ cd terraform/deployments/govuk-test
+   $ terraform apply -var-file=../variables/test/common.tfvars \
+      -var-file=../variables/test/infrastructure.tfvars
+   ```
+
+7. Apply the `deployments/apps` module in your environment
+
+   This will bring up the app's task, defined in step 2.
+   This part should be performed by a Concourse pipeline in future.
```
Comment (fredericfran-gds) [inline docs/adding-an-app.md:58]: Can you link to the concourse pipeline that Richard was showing? Thanks.

Comment (fredericfran-gds) [reply docs/adding-an-app.md:58]: Maybe link this section to the other section in `docs/applying-terraform.md `

File: docs/adding-an-app.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+This document covers the general steps required to bring up a new application
+in the GOV.UK infrastructure (AWS ECS and supporting services).
+
+The steps below may get out of date, so make sure you double check that they
+are still accurate (please correct this doc if you find the steps are inaccurate).
+
+## Steps
+
+1. Add new variables required to `terraform/deployments/variables`
```
Comment (fredericfran-gds) [inline docs/adding-an-app.md:11]: Can the texts in `` be links?

**Review by fredericfran-gds — APPROVED** [review]
Looks, good some minor comments but you can fix and merge anytime.

**Review by sengi — COMMENTED** [review]
Thanks for adding the docs here! It's really helpful to see those developer journeys made visible. ❤️

**Pull request merged**